models/AclStatusCheckService.php

```diff
@@ -0,0 +1,201 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\funcAcl\models;
+
+use core_kernel_classes_Resource;
+use core_kernel_users_Service;
+use InvalidArgumentException;
+use oat\oatbox\service\ServiceManager;
+use oat\oatbox\user\User;
+use oat\oatbox\user\UserService;
+use oat\funcAcl\helpers\CacheHelper;
+use oat\oatbox\service\ConfigurableService;
+use oat\tao\helpers\ControllerHelper;
+use oat\tao\model\accessControl\ActionAccessControl;
+use oat\tao\model\accessControl\data\DataAccessControl;
+use oat\tao\model\accessControl\filter\ParameterFilterProxy;
+
+/**
+ * @author Gabriel Felipe Soares <gabriel.felipe.soares@taotesting.com>
+ *
+ * @TODO Migrate to DI container
+ */
+class AclStatusCheckService extends ConfigurableService
+{
+    /** @var UserService */
+    private $userService;
+
+    /** @var core_kernel_users_Service|null */
+    private $usersService;
+
+    /** @var ActionAccessControl */
+    private $actionAccessControl;
+
+    /** @var ParameterFilterProxy */
+    private $parameterFilter;
+
+    /** @var DataAccessControl */
+    private $dataAccessControl;
+
+    /** @var string[] */
+    private $overloadedControllerAccess;
+
+    /** @var string[] */
+    private $overloadedActionRights;
+
+    public function __construct($options = [])
+    {
+        parent::__construct($options);
+
+        /**
+         * @TODO Use it via DI container after validate the PoC...
+         */
+        $this->userService = ServiceManager::getServiceManager()->get(UserService::SERVICE_ID);
+        $this->usersService = core_kernel_users_Service::singleton();
+        $this->actionAccessControl = ServiceManager::getServiceManager()->get(ActionAccessControl::SERVICE_ID);
+        $this->parameterFilter = new ParameterFilterProxy();
+        $this->dataAccessControl = new DataAccessControl();
+    }
+
+    public function overloadControllerAccess(array $controllerAccess): self
+    {
+        $this->overloadedControllerAccess = $controllerAccess;
+
+        return $this;
+    }
+
+    public function overloadActionRights(array $actionRights): self
+    {
+        $this->overloadedActionRights = $actionRights;
+
+        return $this;
+    }
+
+    public function check(array $filter): array
+    {
+        if (!isset($filter['userId'], $filter['controller'], $filter['action'])) {
+            throw new InvalidArgumentException('Missing required parameters');
+        }
+
+        $user = $this->userService->getUser($filter['userId']);
+        $controller = $filter['controller'];
+        $action = $filter['action'];
+        $requestParameters = $filter['requestParameters'] ?? [];
+        $userRoles = $this->getUserRoles($user);
+
```

Comment:
suggestion: If all these array indexes are mandatory, maybe its values can be passed as standard parameters instead.

```suggestion
    public function check(string $userId, string $controller, string $action, array $requestParameters = []): array
    {
        if (!isset($userId, $controller, $action)) {
            throw new InvalidArgumentException('Missing required parameters');
        }

        $user = $this->userService->getUser($userId);
        $userRoles = $this->getUserRoles($user);

```

Reply:
:+1: